### PR TITLE
Implement code style and unit test checks using GitHub Actions

### PR DESCRIPTION
**Pull request by bennothommo**
This PR converts the Travis CI task that runs unit tests to use GitHub Actions instead. In addition, there is now automated tests for code style to suggest improvements to the quality of code being presented in PRs (based off of [October CMS's code style](https://github.com/octobercms/october/blob/develop/phpcs.xml)).

### CODE REVIEW

**Comment by bennothommo** [comment]
@jaxwilko @LukeTowers when diagnosing the tests failing, I realised that the `UglifyJsFilter` is using the syntax for version 1 of `uglifyjs`. This version is no longer supported (last release is 7 years ago), so I've opted to drop that filter and convert it (and the corresponding tests) to use version 3 - the filter is now called `UglifyJs3Filter`. This is a BC break, so let me know if that's an issue and I'll fix it up.

**Comment by bennothommo** [comment]
The last failure, I believe, is due to us hitting a rate limit with the Google Closure API tests.

**Review by jaxwilko — COMMENTED** [review]
All looks good, nice catch with UglifyJs :)

**Review by jaxwilko — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "assetic",
-  "version": "1.0.0",
+  "version": "2.0.0-dev",
```
Comment (LukeTowers) [inline package.json]: We should be able to just call this 2.0.0
```suggestion
  "version": "2.0.0",
```

File: src/Assetic/Factory/Resource/DirectoryResource.php [hunk 2]
```diff
@@ -73,52 +73,3 @@ protected function getInnerIterator()
...
-        }
-
-        return null === $this->pattern || 0 < preg_match($this->pattern, $name);
-    }
-
-    public function getChildren()
-    {
-        return new self(new \RecursiveDirectoryIterator($this->current()->getPathname(), \RecursiveDirectoryIterator::FOLLOW_SYMLINKS), $this->pattern);
-    }
-}
```
Comment (LukeTowers) [inline src/Assetic/Factory/Resource/DirectoryResource.php:124]: Why was this removed?

Comment (LukeTowers) [reply src/Assetic/Factory/Resource/DirectoryResource.php:124]: Never mind

**Comment by LukeTowers** [comment]
Awesome work @bennothommo, thanks a ton!
